js/lykke2.js

```diff
@@ -0,0 +1,1171 @@
+        //             {
+        //                 "id":"813a3ffa-1c4b-45cb-b13f-1c077ea2748b",
+        //                 "timestamp":1644155923357,
+        //                 "assetPairId":"BCHEUR",
+        //                 "orderId":"1b367978-7e4f-454b-b870-64040d484443",
+        //                 "role":"Taker",
+        //                 "side":"sell",
+        //                 "price":280.569,
+        //                 "baseVolume":0.01,
+        //                 "quoteVolume":2.8056,
+        //                 "baseAssetId":"2a34d6a6-5839-40e5-836f-c1178fa09b89",
+        //                 "quoteAssetId":"EUR",
+        //                 "fee":null
+        //             }
+        //         ],
+        //         "error":null
+        //     }
+        //
+        return this.parseTrades (payload, market, since, limit);
+    }
+
+    parseBidAsk (bidask, priceKey = 0, amountKey = 1) {
+        const price = this.safeString (bidask, priceKey);
+        const amount = Precise.stringAbs (this.safeString (bidask, amountKey));
+        return [ this.parseNumber (price), this.parseNumber (amount) ];
+    }
+
+    async fetchDepositAddress (code, params = {}) {
+        await this.loadMarkets ();
+        const currency = this.currency (code);
+        const request = {
+            'AssetId': this.safeString (currency, 'id'),
+        };
+        const response = await this.privateGetOperationsDepositsAddressesAssetId (this.extend (request, params));
+        //
+        //     {
+        //         "assetId":"2a34d6a6-5839-40e5-836f-c1178fa09b89",
+        //         "symbol":"BCH",
+        //         "address":null,
+        //         "baseAddress":null,
+        //         "addressExtension":null,
+        //         "state":"Active"
+        //     }
+        //
+        const address = this.safeString (response, 'baseAddress');
+        const tag = this.safeString (response, 'addressExtension');
+        this.checkAddress (address);
+        return {
+            'currency': code,
+            'address': address,
+            'tag': tag,
+            'network': undefined,
+            'info': response,
+        };
+    }
+
+    parseTransaction (transaction, currency = undefined) {
+        //
+        // withdraw
+        //     "3035b1ad-2005-4587-a986-1f7966be78e0"
+        //
+        // fetchTransactions
+        //     {
+        //         "operationId":"787201c8-f1cc-45c0-aec1-fa06eeea426b",
+        //         "assetId":"2a34d6a6-5839-40e5-836f-c1178fa09b89",
+        //         "totalVolume":0.1,
+        //         "fee":0.0,
+        //         "type":"deposit",
+        //         "timestamp":1644146723620
+        //     }
+        //
+        let id = undefined;
+        let assetId = undefined;
+        let code = undefined;
+        let amount = undefined;
+        let fee = undefined;
+        let type = undefined;
+        let timestamp = undefined;
+        if (typeof transaction === 'string') {
```

Comment:
i think that instead of doing this we should handle the case for `withdraw` separately, usually we do not call `parseTransaction` on the output of `withdraw`

Reply:
> usually we do not call parseTransaction on the output of withdraw

We actually do `parseTransaction` in `withdraw` in quite a number of cases, just like we do `parseOrder` in `cancelOrder`.